### PR TITLE
Add CORS support for CapSvc; Added ability to change accepted origin at runtime

### PR DESCRIPTION
**Pull request by SEQUOIIA**
Reason: So our Backstage instance can call Capability-Service.

### CODE REVIEW

**Comment by devex-sa** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |4|
| **Quality Gates** |OK|
| **Description** |Low Risk: The change is low risk as it touches fewer modules and code than 33% of your change sets.|
| **Warnings** |Delta Analysis Info<br/><ul><li>2 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping quality gate checks for those files.</li></ul>|

File: src/CapabilityService.WebApi/Properties/launchSettings.json [hunk 1]
```diff
@@ -32,7 +32,8 @@
 			"CAPABILITY_SERVICE_KAFKA_TOPIC_CAPABILITY": "build.selfservice.events.capabilities",
 			"CAPABILITY_SERVICE_KAFKA_TOPIC_TOPICS": "build.selfservice.events.topics",
 			"CAPABILITY_SERVICE_KAFKA_ENABLE_AUTO_COMMIT": "false",
-			"KAFKAJANITOR_API_ENDPOINT": "http://localhost:5000"
+			"KAFKAJANITOR_API_ENDPOINT": "http://localhost:5000",
+			"CAPABILITYSERVICE__CORS__ORIGIN": "http://localhost:3000"
```
Comment (ZaradarBH) [inline src/CapabilityService.WebApi/Properties/launchSettings.json]: Shouldnt we follow the same naming scheme as the other envs? == CAPABILITY_SERVICE_CORS_ORIGIN?

File: src/CapabilityService.WebApi/Startup.cs [hunk 2]
```diff
@@ -48,6 +50,32 @@ public void ConfigureServices(IServiceCollection services)
...
+            {
+	            var apiOptions = new ApiOptions();
+	            Configuration.GetSection(ApiOptions.Position).Bind(apiOptions);
+	            var originHosts = new ArrayList(new[] {"https://build.dfds.cloud", "https://backstage.dfds.cloud" });
+
+	            options.AddDefaultPolicy(builder =>
+	            {
+		            if (apiOptions.Cors != null && apiOptions.Cors.Origin != null)
+		            {
+			            originHosts.Add(apiOptions.Cors.Origin);
```
Comment (ZaradarBH) [inline src/CapabilityService.WebApi/Startup.cs]: It might be a good idea is we could load in multiple hosts via the config?

**Comment by devex-sa** [comment]
[//]: # (4744e426-5795-11e9-8647-d663bd873d93)

|| **CodeScene Delta Analysis Results** |
| --- | ---- |
| **Risk** |4|
| **Quality Gates** |OK|
| **Description** |Low Risk: The change is low risk as it touches fewer modules and code than 33% of your change sets.|
| **Warnings** |Delta Analysis Info<br/><ul><li>2 files in this change set did not exist with the same name on the main branch. This is probably because those files have been renamed or introduced in the commits under analysis. Skipping quality gate checks for those files.</li></ul>|

File: src/CapabilityService.WebApi/Properties/launchSettings.json (hunk 1 above)
Comment (SEQUOIIA) [reply src/CapabilityService.WebApi/Properties/launchSettings.json]: Added!

File: src/CapabilityService.WebApi/Startup.cs (hunk 2 above)
Comment (SEQUOIIA) [reply src/CapabilityService.WebApi/Startup.cs]: Added!

**Pull request closed without merging**